src/handler.rs

```diff
@@ -307,6 +377,24 @@ where
     }
 }
 
+pub struct AsyncResponse<M: AsyncMessage>(M::Result);
+
+impl<A, M> MessageResponse<A, M> for ResponseFuture<AsyncResponse<M>>
+where
+    A: Actor,
+    M: AsyncMessage + 'static,
+    M::Result: Send,
+    A::Context: AsyncContext<A>,
+{
+    fn handle<R: ResponseChannel<M>>(self, _: &mut A::Context, tx: Option<R>) {
+        actix_rt::spawn(async move {
+            if let Some(tx) = tx {
+                tx.send(self.await.0)
+            }
+        });
```

Comment:
I didn't understand the point. When will this code be executed exactly?

Reply:
Maybe I got it. 
But does it matter in this case if actor's context is already waiting on another future to finish?
So, as I understand, it does not matter where this future is executed, because when the waiting is over we any mutable references anywhere.